### PR TITLE
Move/Split/PEP8 test_coordinates

### PR DESCRIPTION
**Pull request by kain88-de**
test_coordinates is now split up into files for each reader class and
some common base classes in base.py. To make sure that this is still
easy to find they now live in their own folder and I have also made
sure that there are almost no PEP8 warning left.

I haven't yet worked on a common base test. Since this PR is already huge I think it is worth it to introduce a common base-reader/writer test in another PR and make sure here that there is no regression in the current tests.

### CODE REVIEW

**Comment by richardjgowers** [comment]
So is `base.py` designed to contain all the reference data?  Because we've got a base module, could we call it `reference` instead?  `from .reference import RefADK`

**Comment by kain88-de** [comment]
It just happened that all the references landed in there. My original thought was to stuff everything in there which was needed from several different reader-classes tests.

But yeah I agree having a reference here is actually nice.

**Comment by kain88-de** [comment]
Ok coverage analysis tells me that I'm missing some tests. It would be nice if someone could help me find the missing tests.

**Comment by richardjgowers** [comment]
I think it's the ImportError stopping tests from going

**Comment by kain88-de** [comment]
Nope the biggest culprit is the `run_in_tempdir` decorator. But I think I found the solution. Instead of decorating the classes I have to inherit from the `tempdir.TempDir` class.

File: testsuite/MDAnalysisTests/test_streamio.py [hunk 1]
```diff
@@ -19,7 +19,7 @@
 import MDAnalysis
 import MDAnalysis.lib.util as util
 import MDAnalysis.tests.datafiles as datafiles
-from MDAnalysisTests.test_coordinates import RefAdKSmall
+from MDAnalysisTests.coordinates.reference import RefAdKSmall
```
Comment (kain88-de) [inline testsuite/MDAnalysisTests/test_streamio.py]: Does anyone have an idea why this fails on travis? I can't reproduce the error on my machine.

Comment (kain88-de) [inline testsuite/MDAnalysisTests/test_streamio.py]: Ok I think I found it. The error was in `setup.py`. It didn't install the new coordinates modules.

**Comment by kain88-de** [comment]
Ok now everything should be working fine. Coverage reports no regressions for me. The coordinates test now also clean up all their temporary files.

**Pull request merged**